Update remote logging feature UI to depend on usage tracking status

chihsuan opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/woocommerce/woocommerce/issues/54824

Improve remote logging feature UI:

1. Adjust the wording to clarify that:
- Remote Logging is only active when tracking is enabled.
- Include clearer messaging for the remote logging feature.

2. Grey Out Setting:
Uncheck and disable (grey out) the Remote Logging setting when tracking is unchecked.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Case1: Opt-out of usage tracking**
1. Create a fresh site with this branch
2. Go to Core Profiler
3. Opt-out of usage tracking and skip core profiler
4. Go to `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
5. Verify that the remote logging feature is unchecked and disabled. The description should mention that usage tracking must be enabled to use remote logging. 
6. Click the link to the usage tracking settings page.
7. Verify that the usage tracking settings page is displayed.
8. Opt-in to usage tracking and go back to the features page.
9. Verify that remote logging is checked.
10. Disable remote logging
11. Disable and re-enable the usage tracking
12. Verify that remote logging is still unchecked.

**Case2: Opt-in of usage tracking**
1. Create a fresh site with this branch
2. Go to Core Profiler
3. Opt-in of usage tracking and skip core profiler
4. Go to `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
5. Verify that the remote logging feature is checked

<img width="923" alt="Screenshot 2025-02-05 at 16 37 42" src="https://github.com/user-attachments/assets/9b37b5aa-2ead-4978-8d1b-7de0156ddda3" />

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

chihsuan committed:
Update remote logging feature to depend on usage tracking status

- Modify remote logging feature description to link to usage tracking settings
- Add conditional logic to disable remote logging when usage tracking is off
- Ensure remote logging is only enabled when usage tracking is active

chihsuan committed:
Add changelog

psealock reviewed (commented):
> 7. Opt-in to usage tracking and go back to the features page.
> 8. Verify that remote logging is unchecked.

Remote logging is checked once I return to the features page

psealock commented on plugins/woocommerce/src/Internal/Features/FeaturesController.php:
```diff
@@ -287,11 +290,13 @@ private function get_feature_definitions() {
+						'</a>'
 					),
-					'enabled_by_default' => true,
+					'enabled_by_default' => WC_Site_Tracking::is_tracking_enabled(),
```
Can you use the `$tracking_enabled` value instead?

chihsuan replied:
Sure! Updated it 438cbc766f9beb0ecf6647269c218d94c346284f.

chihsuan committed:
Update remote logging feature to use tracking status variable

- Replace hardcoded tracking status check with a passed variable
- Maintain existing feature description and settings link
- Improve flexibility of tracking status determination

chihsuan commented:
> Remote logging is checked once I return to the features page

@psealock I wasn't able to reproduce this issue. 🤔 

Here is the video of the steps I took:

https://github.com/user-attachments/assets/51eab319-c0b7-4a15-a63a-c66f2cfc6f88

 Can you confirm if you see the same behavior?

adrianduffell commented:
@chihsuan is it possible to indent the warning icon so it's aligned with the text above. Something like this?

![image](https://github.com/user-attachments/assets/d835bbdb-b636-416c-a4b0-9fc121dbcc24)

chihsuan commented:
> @chihsuan is it possible to indent the warning icon so it's aligned with the text above. Something like this?
> 
> ![image](https://private-user-images.githubusercontent.com/9312929/410392323-d835bbdb-b636-416c-a4b0-9fc121dbcc24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg4MzUxMzcsIm5iZiI6MTczODgzNDgzNywicGF0aCI6Ii85MzEyOTI5LzQxMDM5MjMyMy1kODM1YmJkYi1iNjM2LTQxNmMtYTRiMC05ZmMxMjFkYmNjMjQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMDZUMDk0MDM3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTQzNGE2ZTVhYWVlZWNhMWM0N2MyNTc5NDBjMTQyNTMwMzAxOGQ3MDkyMzY2ZjVkNWNjNjM4YzcyOGZmOTI1ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.p03t1SdhJ_Fv0ulrSelWg6pV_L6ZBHO7nbvAZnT88_c)

@adrianduffell It's possible. However, this will make it look different from other description tips. Do you think it's worth it?

<img width="805" alt="Screenshot 2025-02-06 at 17 42 09" src="https://github.com/user-attachments/assets/eb369589-499f-4054-b888-61771a2ec62c" />

adrianduffell commented:
> However, this will make it look different from other description tips. Do you think it's worth it?

Hmm, I see. I probably think they all should have the indent -- but I don't think it is worth it right now. We can make layout improvements in the modernized settings efforts. Let's leave as-is. Thanks Chi.

psealock reviewed (approved):
>Can you confirm if you see the same behavior?

I checked it on JN twice and the behaviour works as expected. Not sure what happened before, maybe I got it wrong 🤷🏽 

The rest of the behaviour checks out and works as it should, code looks good too 🚢

chihsuan committed:
Enable remote logging by default

ilyasfoo reviewed (approved):
Tests well, LGTM!
